internal/pkg/agent/install/uninstall_windows.go

```diff
@@ -20,22 +20,27 @@ func isBlockingOnExe(err error) bool {
 	if err == nil {
 		return false
 	}
-	path, errno := getPathFromError(err)
```

Comment:
Where did this call go?

Reply:
away - uahuahuahua (sorry I could not resist the joke)

Why it went away:
I don't see any evidence the fact the error does not have a path on it prevents it from being one of the retriable error. Besides the path isn't used at all here, so I don't see the relevance of this check. `err == syscall.ERROR_ACCESS_DENIED` seems to be the only thing relevant to determine if the error `isBlockingOnExe`.

Reply:
Looking at the way this gets used, the error needs to have a path for the special case for removing a blocking exe to work. The call is unnecessary in the overall logic chain since it gets called again later, but if the error doesn't have a path the work around for it can't work.

Reply:
I wonder if SHARING_VIOLATION was just forgotten here.

Reply:
> I wonder if SHARING_VIOLATION was just forgotten here.
me too.

However the error in the test doe not come from `isBlockingOnExe`, but from `isRetryableError`. So that might be only checking for `syscall.ERROR_ACCESS_DENIED` intentionally.

>Looking at the way this gets used, the error needs to have a path for the special case for removing a blocking exe to work. The call is unnecessary in the overall logic chain since it gets called again later, but if the error doesn't have a path the work around for it can't work.

yes, but it's another case and there it still checks for the path not being empty.

Reply:
Reading the original PR https://github.com/elastic/elastic-agent/pull/3120 the implemetation was adapted from https://github.com/golang/go/blob/master/src/testing/testing.go#L1257 which does the same.

> yes, but it's another case and there it still checks for the path not being empty.

If we get rid of the path check here, this function is no longer answering the question "is this an error from blocking waiting on an exe" it is just "is this an access denied error" which is mislead. Just because the code happens to be written today so that this doesn't matter, doesn't mean that will stay true in the future.

You could rename the function to something like `possiblyBlockingOnExe` to indicate that the error here might not have anything to do with an executable file. There are a lots of other reasons to get an access is denied error.

Reply:
Well either way it does not answer _"is this an error from blocking waiting on an exe"_ as it did not do any check on the path, not even if it was from an exe file.
Also the Go implementation does not check the path. I'm happy to rename it, but unless there is a proper check on the path, I don't believe to be worth checking it